src/App.tsx

```diff
@@ -6,39 +6,48 @@ import './App.scss';
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
-import { UserSelector } from './components/UserSelector';
+import { UserSelector } from './components/UserSelector/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
 import { User } from './types/User';
-import { Post } from './types/Post';
+import { getUsers } from './api/users';
+import { usersSlice } from './store/usersSlice';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { postsSlice } from './store/postsSlice';
+import { selectedPostSlice } from './store/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
   const [loaded, setLoaded] = useState(false);
   const [hasError, setError] = useState(false);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const posts = useAppSelector(state => state.posts.posts);
```

Comment:
[Not critical]
It makes sense to define selector callback outside of the components, along with reducers/slices. For example:
```
// reducer.ts
const postsSelector = state => state.posts.posts;
export { postsSelector };

// component
import { postsSelector } from "./reducer.ts";
...
const posts = useAppSelector(postsSelector);
```
With this approach, knowledge of store structure is isolated within reducer module. So, if you'd have to restructure the store, you'd only have to make the changes inside reducer module and not in the components which use data from it.